data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaCustomConsumer.java

```diff
@@ -452,7 +452,7 @@ private void processRecord(final AcknowledgementSet acknowledgementSet, final Re
         // buffer contents before the event record is added
         // to acknowledgement set
         if (acknowledgementSet != null) {
-            acknowledgementSet.add(record.getData());
+            acknowledgementSet.add(((Event)record.getData()).getEventHandle());
```

Comment:
If we continue to support `add(Event)` then all of this code should be simpler and you can remove for this PR.